### ISSUE TITLE
T-SQL temp tables throw error for weird syntax

### ISSUE THREAD

**Issue by pwildenhain** [sqlfluff/sqlfluff#5571]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I'm getting a parsing error on some terrible T-SQL

### Expected Behaviour

It would be great if SQLFluff didn't break on my terrible SQL

### Observed Behaviour

`test.sql`

```sql
select column_1 from ."#my_table"
```

```sh
$ sqlfluff lint test.sql --dialect tsql
== [test.sql] FAIL                                                                                         
L:   1 | P:  17 |  PRS | Line 1, Position 17: Found unparsable section: 'from
                       | ."#my_table"'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

Believe it or not this is valid SQL 😅 

The issue is the dot (`.`) before `"#my_table` if I drop that, then everything works. However, this file is auto-generated from a dbt ephemeral model, so I'm not sure that's possible...

### How to reproduce

Run the above example

### Dialect

T-SQL

### Version

sqlfluff, version 2.3.5
Python 3.12.1

### Configuration

NA

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by pwildenhain** [comment]
I'm guessing that the change needs to be made somewhere here?

https://github.com/sqlfluff/sqlfluff/blob/0591b0ace98075502cc267383fdfbc5636f7477e/src/sqlfluff/dialects/dialect_tsql.py#L3938-L3962
